### ISSUE TITLE
time.Format - timezone details missing with time_full and time_long with TOML dates 

### ISSUE THREAD

**Issue by jmooring** [gohugoio/hugo#8895]
This issue title is... terrible.

content/tests/pass.md

```toml
+++
title = "Pass"
date = "2021-08-16T06:00:00-07:00"  # String
draft = false
+++
```

content/tests/fail.md

```toml
+++
title = "Fail"
date = 2021-08-16T06:00:00-07:00    # Date
draft = false
+++
```

||pass.md|fail.md
:--|:--|:--
`{{ .Date \| time.Format ":time_full" }}`|6:00:00 am Pacific Daylight Time|6:00:00 am
`{{ .Date \| time.Format ":time_long" }}`|6:00:00 am PDT|6:00:00 am

**Comment by bep** [comment]
I suspect this is a duplicate of #8891

**Comment by jmooring** [comment]
> hugo v0.88.0-DEV-EFEBD756+extended linux/amd64 BuildDate=2021-08-16T10:10:04-0700

Includes fix for #8891.

**Comment by bep** [comment]
Hmm, I'm added a test for this re. some related issue, and it ... fails. Not sure what happens, but it reminds me to never commit fixes for issues without a test.

**Comment by bep** [comment]
I'm reopening this as I tried to add the above as a test case, but it fails.

I have removed the patch that was supposed to fix this, as I didn't understand it and it came without a test case.

Note that I'm not sure this should/could be fixed:

* go-toml parses offset dates on RFC 3339 format, with empty zone name (expect for UTC, oddly enough)
* The way Hugo parses RFC 3339 is a little differently, but in my tests I get all empty zone name in this case, too. I suspect that this may be computer dependant.
